virtool/account/api.py

```diff
@@ -195,8 +201,15 @@ async def post(
 
         user_id = self.request["client"].user_id
 
+        data_dict = data.dict(exclude_unset=True)
+
+        permissions_dict = {}
+
+        if "permissions" in data_dict:
+            permissions_dict = data.permissions.dict(exclude_unset=True)
```

Comment:
Can't this just be:
```suggestion
        permissions_dict = data.permissions.dict(exclude_unset=True)
```
The `permissions_dict` variable could also be dropped if the above works.

Reply:
If permissions is not provided when creating a key I believe this would result in an error.

Reply:
I have just been working on another issue with this problem.

Calling `EditAccountSchema.dict(exclude_unset=True)` will actually recurse down into any nested models. This means `data.permissions` is turned into a `dict` and permissions not explicity provided in the request are excluded.

Reply:
https://github.com/virtool/virtool/blob/52d9b08645a7cc024c49144e17475fdb0b29368b/virtool/groups/data.py#L72

Reply:
Would this not result in an error if permissions itself is not provided in the request though? For example, in TestCreateAPIKey the body could be as below:

        body = {"name": "Foobar"}

        resp = await client.post("/account/keys", body)

Then data.permissions.dict gives an error. Maybe a try: except block would be better?

Also, if data["permissions"] is being used in your example, should we use data_dict["password"] after all instead of data.password to be consistent?

Reply:
It's not necessary to call `data.permissions.dict()` calling `data.dict(exclude_unset=True)` converts `permissions` to a `dict` if it exists. Everything gets converted to a `dict`.